pkg/operator/hostendpointscontroller/host_endpoints_controller.go

```diff
@@ -0,0 +1,334 @@
+	eventRecorder        events.Recorder
+	queue                workqueue.RateLimitingInterface
+	cachesToSync         []cache.InformerSynced
+	operatorClient       v1helpers.OperatorClient
+	infrastructureLister configv1listers.InfrastructureLister
+	podLister            corev1listers.PodLister
+	endpointsLister      corev1listers.EndpointsLister
+	endpointsClient      corev1client.EndpointsGetter
+}
+
+func NewHostEndpointsController(
+	operatorClient v1helpers.OperatorClient,
+	eventRecorder events.Recorder,
+	kubeClient kubernetes.Interface,
+	kubeInformersForNamespaces operatorv1helpers.KubeInformersForNamespaces,
+	infrastructureInformer configv1informers.InfrastructureInformer,
+) *HostEndpointsController {
+	kubeInformersForTargetNamespace := kubeInformersForNamespaces.InformersFor("openshift-etcd")
+	endpointsInformer := kubeInformersForTargetNamespace.Core().V1().Endpoints()
+	podInformer := kubeInformersForTargetNamespace.Core().V1().Pods()
+
+	c := &HostEndpointsController{
+		eventRecorder: eventRecorder.WithComponentSuffix("host-etcd-endpoints-controller"),
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "HostEndpointsController"),
+		cachesToSync: []cache.InformerSynced{
+			operatorClient.Informer().HasSynced,
+			endpointsInformer.Informer().HasSynced,
+			podInformer.Informer().HasSynced,
+			infrastructureInformer.Informer().HasSynced,
+		},
+		operatorClient:       operatorClient,
+		infrastructureLister: infrastructureInformer.Lister(),
+		podLister:            podInformer.Lister(),
+		endpointsLister:      endpointsInformer.Lister(),
+		endpointsClient:      kubeClient.CoreV1(),
+	}
+	operatorClient.Informer().AddEventHandler(c.eventHandler())
+	endpointsInformer.Informer().AddEventHandler(c.eventHandler())
+	infrastructureInformer.Informer().AddEventHandler(c.eventHandler())
+	podInformer.Informer().AddEventHandler(c.eventHandler())
+	return c
+}
+
+func (c *HostEndpointsController) sync() error {
+	err := c.syncHostEndpoints()
+
+	if err != nil {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "HostEndpointsDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "ErrorUpdatingHostEndpoints",
+			Message: err.Error(),
+		}))
+		if updateErr != nil {
+			c.eventRecorder.Warning("HostEndpointsErrorUpdatingStatus", updateErr.Error())
+			return updateErr
+		}
+		return err
+	}
+
+	_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+		Type:   "HostEndpointsDegraded",
+		Status: operatorv1.ConditionFalse,
+		Reason: "HostEndpointsUpdated",
+	}))
+	if updateErr != nil {
+		c.eventRecorder.Warning("HostEndpointsErrorUpdatingStatus", updateErr.Error())
+		return updateErr
+	}
+	return nil
+}
+
+func (c *HostEndpointsController) syncHostEndpoints() error {
+	discoveryDomain, err := c.getEtcdDiscoveryDomain()
```

Comment:
Wild idea Luis.  Does this still work if you put a bunch of bad values in?  If so, can we simply put the internal IP addresses of all the master nodes and the dns name of the bootstrap node and just always keeps it like that?

Reply:
There should be no problem putting the actual IP addresses in there. As long as kube-apiserver's etcd client handle failing over non-ready nodes.